File: frontend/src/app/features/caisse/services/caisse.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { API_BASE_URL } from '../../../core/services/api.config';
import { CaisseCreateRequest } from '../models/caisse-create-request';
import { CaisseResponse } from '../models/caisse-response';

@Injectable({
  providedIn: 'root'
})
export class CaisseService {
  private readonly apiUrl = `${API_BASE_URL}/caisses`;

  constructor(private http: HttpClient) {}

  getAll(): Observable<CaisseResponse[]> {
    return this.http.get<CaisseResponse[]>(this.apiUrl);
  }

  getActives(): Observable<CaisseResponse[]> {
    return this.http.get<CaisseResponse[]>(`${this.apiUrl}/actives`);
  }

  getAccessibles(): Observable<CaisseResponse[]> {
    return this.http.get<CaisseResponse[]>(`${this.apiUrl}/accessibles`);
  }

  getById(id: number): Observable<CaisseResponse> {
    return this.http.get<CaisseResponse>(`${this.apiUrl}/${id}`);
  }

  create(request: CaisseCreateRequest): Observable<CaisseResponse> {
    return this.http.post<CaisseResponse>(this.apiUrl, request);
  }

  initialiserMaCaisse(): Observable<CaisseResponse> {
    return this.http.post<CaisseResponse>(`${this.apiUrl}/initialiser-ma-caisse`, {});
  }
}

File: backend/src/main/java/com/mini/credit/service/CaisseService.java
package com.mini.credit.service;


import com.mini.credit.dto.caisse.CaisseCreateRequest;
import com.mini.credit.dto.caisse.CaisseResponse;

import java.util.List;

public interface CaisseService {
    CaisseResponse create(CaisseCreateRequest request);
    CaisseResponse initialiserMaCaisse();
    CaisseResponse getById(Long id);
    List<CaisseResponse> getAll();
    List<CaisseResponse> getActives();
    List<CaisseResponse> getAccessibles();
}